pkg/collector/corechecks/servicediscovery/usm/service.go

```diff
@@ -125,6 +125,18 @@ func abs(p string, cwd string) string {
 	return path.Join(cwd, p)
 }
 
+func getExecutableBasename(exe string) string {
+	suffixIdx := strings.IndexFunc(exe, func(r rune) bool {
+		return r == '.' || unicode.IsDigit(r)
+	})
+
+	if suffixIdx < 0 {
+		return exe
+	}
+
+	return exe[:suffixIdx]
+}
```

Comment:
name := strings.TrimSuffix(basename, filepath.Ext(basename))

Reply:
I think the function needs to be documented as the name suggests different intention than the code

Reply:
> I think the function needs to be documented as the name suggests different intention than the code

Maybe "basename" is too synonym with "strip the extension"; I am open for suggestions for a better name for what this function returns. I'll also add a comment with examples of what it is doing.

> name := strings.TrimSuffix(basename, filepath.Ext(basename))

This function does more than stripping the file extension. Examples:
- python3 -> python
- python3.10 -> python
- dotnet.exe -> dotnet
- python3-10.exe -> python